app/javascript/components/public_health/CustomExport.js

```diff
@@ -0,0 +1,508 @@
+            };
+          }),
+        },
+      },
+    };
+  }
+
+  componentDidMount() {
+    // get patient count preview if this is a preset
+    if (this.props.preset?.id) {
+      this.getPatientCount();
+    }
+  }
+
+  // Save a new preset
+  save = () => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .post(`${window.BASE_PATH}/user_export_presets`, this.state.preset)
+      .catch(() => toast.error('Failed to save export preset.'))
+      .then(response => {
+        if (response?.data) {
+          toast.success('Export preset successfully saved.');
+          this.handlePresetChange('id', response?.data?.id);
+        }
+        this.props.reloadExportPresets();
+      });
+  };
+
+  // Update an existing preset
+  update = () => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .put(`${window.BASE_PATH}/user_export_presets/${this.state.preset?.id}`, this.state.preset)
+      .catch(() => toast.error('Failed to update export preset.'))
+      .then(response => {
+        if (response?.data) {
+          toast.success('Export preset successfully updated.');
+        }
+        this.props.reloadExportPresets();
+      });
+  };
+
+  // Delete an existing filter
+  delete = () => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .delete(`${window.BASE_PATH}/user_export_presets/${this.state.preset?.id}`)
+      .catch(() => toast.error('Failed to delete export preset.'))
+      .then(() => {
+        toast.success('Export preset successfully deleted.');
+        this.handlePresetChange('id', null);
+        this.props.reloadExportPresets();
+      });
+  };
+
+  // Export data with current configurations
+  export = () => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .post(`${window.BASE_PATH}/export/custom`, this.state.preset)
+      .then(response => {
+        if (response?.status === 200) {
+          toast.success('Export has been initiated!');
+          this.props.onClose();
+        }
+      })
+      .catch(err => {
+        reportError(err?.response?.data?.message ? err.response.data.message : err, false);
+        this.props.onClose();
+      });
+  };
+
+  // Get patient count based on custom query
+  getPatientCount = () => {
+    // cancel any previous unfinished requests to prevent race condition inconsistencies
+    this.state.cancel_token.cancel();
+
+    // generate new cancel token for this request
+    const cancel_token = axios.CancelToken.source();
+
+    this.setState({ cancel_token }, () => {
+      axios
+        .post(`${window.BASE_PATH}/public_health/patients/count`, {
+          query: this.state.custom_patient_query,
+          cancelToken: this.state.cancel_token.token,
+        })
+        .then(response => this.setState({ filtered_monitorees_count: response?.data?.count }))
```

Comment:
I would add an if check here rather than potentially setting `filtered_monitorees_count` to nil if `response` doesn't have data or data doesn't have count for whatever reason.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/562/commits/1f13143f8968feff3c7e9ada6139b47e5cd4ecff#diff-14793b3bc71143b45317a3d08474d370db9ffdfb9643e754341eeda1ab5f096bR130